### PR TITLE
Launcher refactor

### LINKED ISSUE #2330: Launcher restructure
- Remove the launcher side item.
- Add a `File -> New` and a single button in the file browser that brings up a launcher tab.
- Remove other `New` items in the `File` menu.
- Show launcher items for each combination of notebook/console and kernel name.
  -- The order will be updated based on most recently used.
- Replace the launcher tab with the selected item with no additional dialog.
- Add a `File -> Rename`.
- Merge the landing page as a launcher with additional information displayed.

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #2330.

- [x] Remove landing extension and launcher side bar.
- [x] Update launching behavior for terminal, console, notebook, and editor. 
- [x] Replace launcher tab with the opened widget.  
- [x] Removed the palette items for new text file and notebook.
- [x] Create menu item for `New...`.
- [x] Properly sort items.
- [x] Update filebrowser buttons to use toolbar and call launcher.
- [x] Add a `File -> Rename` with dialog.
- [x] Add a right click handler on file titles with rename.
- [x] Add logo and version if displayed as "landing page".
- [x] Pick convenient keybinding for the new action.

<image src="https://user-images.githubusercontent.com/2096628/26892830-26d1e550-4b7f-11e7-8325-5cbd8c6c5bf8.png" width=500>

### CODE REVIEW

**Comment by ellisonbg** [comment]
Let me know when this is ready for review...no hurry, just willing to help in any way I can

**Comment by blink1073** [comment]
Update: the launching behavior is done for terminal, console, notebook, and editor.  The launcher tab is replaced with the opened widget.

**Comment by blink1073** [comment]
@ian-r-rose, FYI I am about to use expose a toolbar in the file browser.

**Comment by ian-r-rose** [comment]
Awesome, I will abandon a branch that does that!

**Comment by ellisonbg** [comment]
Did an initial UX review of this:

* Very, very cool overall, will be a huge improvement.
* Having a keyboard shortcut for "New" will make it even easier.
* We may want to have a CSS class that is applied to "Sections" of activities (notebooks, consoles, etc). That way, we can do conditional styling/design later.
* Eventually (post beta) we may want to allow visual sectionings of the items, or different ways of sorting the items. At least having a notion of "setion" in the API would be a help at this point (you may already have it in the code, I haven't looked yet).

We can have @cameronoelsen take a pass on the design side once this is merged, so I wouldn't spend too much time on that side of it - already though, the visual design is super solid and usable.

Will you need a "new folder" icon for the filebrowser toolbar? if so, @cameronoelsen can you create one? I know you had worked on that before so it may be mostly done.

This is fantastic!

**Comment by ellisonbg** [comment]
Also, how are you sorting the items?

**Comment by blink1073** [comment]
We have a `section`, but no associated class applied yet.  We don't yet have icons for new file or new folder (nor are they in font-awesome), so I was going to use a single `+` and add a `New Folder` item in the file menu.  

We are currently sorting by `rank`, then `section`, then `displayName`.

**Comment by ian-r-rose** [comment]
Does the launcher make a distinction between file-backed activities and non-file-backed activities?

**Comment by blink1073** [comment]
No, they are treated the same.

**Comment by blink1073** [comment]
I put a placeholder in for now for the new folder, since `File -> New Folder` would only apply to the main file browser.

<image src="https://user-images.githubusercontent.com/2096628/26901076-24071a64-4b9a-11e7-9d63-23e7588b6b87.png" width=300>

**Comment by blink1073** [comment]
Rename dialog:

<image src="https://user-images.githubusercontent.com/2096628/26903513-2a9b7542-4ba3-11e7-8930-dd0695567032.png" width=300>

**Comment by blink1073** [comment]
Variations of `Accel T` and `Accel N` are intercepted by the browser:

<image src="https://user-images.githubusercontent.com/2096628/26904542-05b8fb1e-4ba8-11e7-85a0-289003b52b35.png" width=300>

**Comment by blink1073** [comment]
@ellisonbg this is ready for review, I can add the "landing" page behavior in a follow on PR.

**Comment by blink1073** [comment]
Finished the landing page (shown after app restore when there are no open widgets):

<image src="https://user-images.githubusercontent.com/2096628/26927267-c00de1aa-4c16-11e7-9428-edc99c18ce4b.png" width=400>

**Comment by ellisonbg** [comment]
OK, reviewing now!

File: packages/launcher/src/index.ts [hunk 1]
```diff
@@ -99,26 +102,69 @@ interface ILauncher {
...
+   * The default value is the displayName.
+   */
+  name?: string;
+
+  /**
+   * The category for the launcher item.
+   *
+   * The default value is the an empty string.
+   */
+  category?: string;
```
Comment (ellisonbg) [inline packages/launcher/src/index.ts]: The only think I can imagine adding here later is a field for the kernel name. That could let us sort by category or kernel later. But we can always add that later.

**Pull request merged**